### ISSUE TITLE
ProcessPool default timeout

### ISSUE THREAD

**Issue by csm10495** [noxdafox/pebble#145]
Hey folks,

Could we add a parameter to the initializer for ProcessPool to take a timeout value? It would default to None to have the same behavior as today.

If calling submit/schedule without a timeout parameter (or set to None) it would use the default timeout set by the initializer, otherwise would use the one passed in.

This would make it easy to have a pool have a default timeout for tasks.

**Comment by csm10495** [comment]
If its a reasonable idea, I'd be happy to file a PR too :)

**Comment by noxdafox** [comment]
Hello,

I would refrain to spread too many redundant parameters in the API. Goal is to keep the API clean and simple.

Although it might sound convenient, redundancy is an anti-patter when designing APIs. It introduces ambiguity (what if you specify `timeout` in both places?) and makes the control-flow hard to follow.

If you need such a functionality, you can easily achieve it with few lines of Python:
```python
import pebble

class DefaultTimeoutPool(pebble.ProcessPool):
    def __init__(self, *args, timeout=None, **kwargs):
        super().__init__(*args, **kwargs)
        self._timeout = timeout

    def schedule(self, *args, **kwargs):
        if 'timeout' not in kwargs or kwargs['timeout'] is None:
            kwargs['timeout'] = self._timeout
            super().schedule(*args, **kwargs)
```

**Comment by csm10495** [comment]
Makes sense. Thanks still.
